contracts/utils/Votes.sol

```diff
@@ -0,0 +1,206 @@
+
+    bytes32 private constant _DELEGATION_TYPEHASH =
+        keccak256("Delegation(address delegatee,uint256 nonce,uint256 expiry)");
+    mapping(address => address) private _delegateCheckpoints;
+    mapping(address => Checkpoints.History) private _userCheckpoints;
+    mapping(address => Counters.Counter) private _nonces;
+    Checkpoints.History private _totalCheckpoints;
+
+    /**
+     * @dev Emitted when an account changes their delegate.
+     */
+    event DelegateChanged(address indexed delegator, address indexed fromDelegate, address indexed toDelegate);
+
+    /**
+     * @dev Emitted when a token transfer or delegate change results in changes to an account's voting power.
+     */
+    event DelegateVotesChanged(address indexed delegate, uint256 previousBalance, uint256 newBalance);
+
+    /**
+     * @dev Returns total amount of votes for account.
+     */
+    function getVotes(address account) public view virtual override returns (uint256) {
+        return _userCheckpoints[account].latest();
+    }
+
+    /**
+     * @dev Returns total amount of votes at given blockNumber.
+     */
+    function getPastVotes(address account, uint256 blockNumber) public view virtual override returns (uint256) {
+        return _userCheckpoints[account].getAtBlock(blockNumber);
+    }
+
+    /**
+     * @dev Retrieve the `totalVotingPower` at the end of `blockNumber`. Note, this value is the sum of all balances.
+     * It is but NOT the sum of all the delegated votes!
+     *
+     * Requirements:
+     *
+     * - `blockNumber` must have been already mined
+     */
+    function getPastTotalSupply(uint256 blockNumber) public view virtual override returns (uint256) {
+        require(blockNumber < block.number, "ERC721Votes: block not yet mined");
+        return _totalCheckpoints.getAtBlock(blockNumber);
+    }
+
+    /**
+     * @dev Returns total amount of votes.
+     */
+    function _getTotalVotes() internal view virtual returns (uint256) {
+        return _totalCheckpoints.latest();
+    }
+
+    /**
+     * @dev Get number of checkpoints for `account` including delegation.
+     */
+    function _getTotalAccountVotes(address account) internal view virtual returns (uint256) {
+        return _userCheckpoints[account].length();
+    }
+
+    /**
+     * @dev Delegate votes from the sender to `delegatee`.
+     */
+    function delegate(address delegatee) public virtual override {
+        address delegator = _msgSender();
+        _delegate(delegator, delegatee);
+    }
+
+    /**
+     * @dev Returns account delegation.
+     */
+    function delegates(address account) public view virtual override returns (address) {
+        return _delegateCheckpoints[account];
+    }
+
+    /**
+     * @dev Change delegation for `delegator` to `delegatee`.
+     *
+     * Emits events {DelegateChanged} and {DelegateVotesChanged}.
+     */
+    function _delegate(address delegator, address newDelegation) internal virtual {
```

Comment:
I'd put that after `delegateBySig` 

with my previous comment, that would result in the following:
```
function delegates(address) public view virtual override returns (address); // public view
function delegate(address) public virtual override;
function delegateBySig(address, uint256, uint256, uint8 v, bytes32 r, bytes32 s) public virtual override;
function _delegate(address, address) internal virtual;
```